source/standards/test-driven-development.html.md.erb

```diff
@@ -0,0 +1,36 @@
+---
+title: Test-driven development (TDD)
+last_reviewed_on: 2024-10-21
+review_in: 6 months
+---
+# <%= current_page.data.title %>
+
+GDS advocates for agile software development practices because we believe that flexibility and responsiveness-to-change are key to building software that meets the needs of our users.
+
+Test-driven development (TDD) is a core agile software development practice which aims to build flexibility and correctness into production software.
+
+Kent Beck, the author of Extreme Programming Explained, describes [TDD][canon-tdd] as .
+
+The process starts with a set of desired behaviours for a piece of code. This list could include happy-path behaviours, error cases, and interaction points with other components.
+
+The inner-loop of TDD is also commonly referred to as "Red-Green-Refactor"; red to represent a failing test, green to represent all passing tests, and refactoring only when all tests are passing.
+
+<%= partial 'partials/svg-tdd-red-green-refactor' %>
+
+### Red
+Turn exactly one item on the list into an actual, concrete, runnable test. This test should fail when you run it.
```

Comment:
It might not be worth complicating this, but a possible addendum to this is "it should fail for the reason you think it should fail" i.e. not compilation errors, not some unexpected reason that might mean you've made a mistake in writing the test.

Reply:
Good shout - I remember someone at my previous job naming this "calling your shots", and that unexpected failures were a sign of an implementation gap and also a good learning opportunity. 😂